packages/swing-store-lmdb/test/test-state.js

```diff
@@ -37,15 +42,30 @@ function testStorage(t, storage) {
 }
 
 test('storageInLMDB', t => {
+  t.equal(isSwingStore('testdb'), false);
```

Comment:
Not sure what this line is actually testing.  It is permissible (in fact, I rely on it a lot during development) to do an `initSwingStore` to reinitialize an existing store.  It is also permissible to `openSwingStore` on an empty directory or non-existent directory, where it will just be treated as an empty store (relevant files and directories will appear as writes happen).  This is also useful.  The more I think about it, the less convinced I am that the `isSwingStore` predicate is of much use at all.

Reply:
My plan is to have cosmic-swingset's `launch-chain.js`, which is supposed to only be run after `reset-state.js` has constructed an initialized store, do something like:

```js
if (simple.isSwingStore(basedir)) {
  storage = simple.openSwingStore(basedir);
} else if (lmdbstore.isSwingStore(basedir)) {
  storage = lmdbstore.openSwingStore(basedir);
} else {
  throw Error(`how the heck did you initialize ${basedir}??`);
}
```

I intended `isSwingStore(dir)` to mean "is `dir` *already* a store of the type that you know how to open?", so I'm fine if it returns `false` on a missing/empty directory. But that particular check isn't important for my use case, and I'll remove it if you prefer. It's the other two checks that matter for me, so that a SimpleStore rejects a directory set up by LMDB, and vice versa.

(the idea is to pass some sort of `--db=simple` flag into the argv of `init-chain.js`, and specifically *avoid* needing to pass one into `launch-chain.js`, to remove the error case where someone might init with LMDB but then run with Simple)

Reply:
I think I see -- lemme try this on for size: the intended use is in a context where there is a presumption of an existing swing store, at which point the only question is: what kind of store is it?  In particular, in such a context it would likely be some kind of error for the store not to exist at all.  This makes sense to me as a use case, though I think this suggests that the test you created is slightly off the mark because it only coincidentally has the correct precursor state for the `isSwingStore` predicate to be meaningful in the way you intend.

Reply:
Yep, that was the intention. If cosmic-swingset somehow got all the way to launch-chain without having a recognizable storage directory in place, we should be able to recognize that and explode appropriately.

I'm guessing that my test additions are incorrectly assuming that the testdir will be missing when the test begins, since that onExit deletion/cleanup doesn't necessarily run between tests (I don't know what the syntax is). How about I remove that first (supposedly-empty directory) assertion?

Reply:
That would work.  Alternatively you could delete the directory at the start of the test (obviously ignoring the failure if the directory already doesn't exist).

Reply:
ooh, I like that better

Reply:
ok how's 4900070 look?